Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs

```diff
@@ -116,6 +120,16 @@ updatePerson personId req = do
   mbEncEmail <- encrypt `mapM` req.email
 
   refCode <- join <$> validateRefferalCode personId `mapM` req.referralCode
+  whenJust req.disability $ \disability -> do
+    customerDisability <- B.runInReplica $ PDisability.findByPersonId personId
+    when (isNothing customerDisability) $ do
+      newDisability <- makeDisability disability
+      PDisability.create newDisability
+      QPerson.updateHasDisability personId
+    when (isJust customerDisability) $ do
+      let disabilityId = getId $ disability.disabilityId
+      PDisability.updateDisabilityByPersonId personId disabilityId disability.tag disability.description
+      QPerson.updateHasDisability personId
```

Comment:
Will already be true, right?

Reply:
Also, please one function for all this and use it here.

Reply:
@0utkarsh I didn't get you, Can you please elaborate

Reply:
You already checked - if the person had a disability stored previously, you are only updating it. But since a disability was already stored, `hasDisability` would be true as well, so no need to mark it true again.

Reply:
```
when (isJust customerDisability) $ do
      QPerson.updateHasDisability personId
```

When condition guarantees that the customer had a disability which means his flag in person table was already set to true.